client/packages/common/src/plugins/components/PluginProvider.tsx

```diff
@@ -1,71 +1,63 @@
-import React, {
-  FC,
-  PropsWithChildren,
-  useState,
-  useMemo,
-  useEffect,
-} from 'react';
 import {
-  PluginContext,
-  PluginControllerState,
-  PluginState,
-} from './PluginContext';
-import { Plugin, PluginArea, PluginType } from '../types';
-import { PluginLoader } from './PluginLoader';
+  ColumnPlugin,
+  ComponentPlugin,
+  create,
+  ArrayUtils,
+} from '@openmsupply-client/common';
 
-interface PluginProviderProps {
-  plugins: Plugin<unknown>[];
-}
-
-export const PluginProvider: FC<PropsWithChildren<PluginProviderProps>> = ({
-  children,
-  plugins,
-}) => {
-  const [pluginState, setState] = useState<PluginState>({
-    plugins: [],
-  });
-
-  const pluginController: PluginControllerState = useMemo(
-    () => ({
-      setState,
-      setPlugins: (plugins: Plugin<unknown>[]) =>
-        setState(state => ({ ...state, plugins })),
-      getPlugins: (area: PluginArea, type: PluginType) =>
-        plugins.filter(plugin => plugin.area === area && plugin.type === type),
-      getPluginElements: function <T>({
-        area,
-        type,
-        data,
-      }: {
-        area: PluginArea;
-        type: PluginType;
-        data?: T;
-      }) {
-        return (plugins as Plugin<T>[])
-          .filter(plugin => plugin.area === area && plugin.type === type)
-          .map(plugin => (
-            <PluginLoader
-              name={plugin.name}
-              module={plugin.module}
-              data={data}
-              path={plugin.path}
-              key={`${plugin.name}-${plugin.module}`}
-            />
-          ));
-      },
-
-      ...pluginState,
-    }),
-    [setState, pluginState]
-  );
-
-  useEffect(() => {
-    setState({ ...pluginState, plugins });
-  }, [plugins]);
-
-  return (
-    <PluginContext.Provider value={pluginController}>
-      {children}
-    </PluginContext.Provider>
-  );
+type PluginProvider = {
+  columnPlugins: ColumnPlugin[];
+  componentPlugins: ComponentPlugin[];
+  addColumnPlugin: (plugin: ColumnPlugin) => void;
+  updateColumnPlugin: (plugin: ColumnPlugin) => void;
+  addComponentPlugin: (plugin: ComponentPlugin) => void;
+  getComponentPlugins: <T extends ComponentPlugin['type']>(
+    type: T
+  ) => Extract<ComponentPlugin, { type: T }>[];
+  getColumnPlugins: <T extends ColumnPlugin['type']>(
+    type: T
+  ) => Extract<ColumnPlugin, { type: T }>[];
 };
+
+export const usePluginProvider = create<PluginProvider>((set, get) => {
```

Comment:
changed to use `zustand` rather than a react Context.  it makes it slightly more awkward to use, because it has to be called with the hook - this is from Andrei's recommendation

Reply:
Why do you prefer zustand? is it just for better performance?

Reply:
that was Andrei's preference, I'm not sure if performance is better or not tbh